### ISSUE TITLE
Not able to connect over bastion host in coreos

### ISSUE THREAD

**Issue by alexraju** [hashicorp/terraform#7196]
Hi there,

### Terraform Version
Terraform v0.6.16

### Affected Resource(s)
- aws_instance

### Terraform Configuration Files
```
provider "aws" {
  access_key = "XXXXXXX"
  secret_key = "XXXXXXXXXXXXXXXX"
  region = "ap-northeast-1"
}

resource "aws_instance" "test" {
  ami = "ami-962c39f8"
  availability_zone = "ap-northeast-1a"
  instance_type = "m3.2xlarge"
  key_name = "Store_Production"
  security_groups = [
    "sg-f964109d"]
  subnet_id = "subnet-cca32hba"
  source_dest_check = false

  connection {
    user = "core"
    agent = false
    private_key = "./private_key.pem"
    bastion_host = "52.22.144.23"
    bastion_private_key = "./private_key.pem"
  }

  tags {
    Name = "test-terraform"
  }

  provisioner "local-exec" {
    command = "echo test creation"
  }

  provisioner "file" {
    source = "./test-file.sh"
    destination = "/tmp/test-file.sh"
  }
```

### Debug Output
https://gist.github.com/alexraju/a275cd8b62beaeccb19f25423a53767c

### Expected Behavior
Connect to the test aws instance using bastion host 

### Actual Behavior
Got error `* Error connecting to bastion: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain`

### Steps to Reproduce
Run `terraform apply` with the above hal code

### Important Factoids
Got the issue only on CoreOS. Tried the same code replacing with CentOS ami (ami-eec1c380) and connection->user value "centos" and got connected and the temp file got transferred.

**Comment by jen20** [comment]
Hi @alexraju! Thanks for opening an issue. Can you confirm whether or not you are able to SSH into the bastion box from outside of Terraform using the keys you specify in `var.aws_key_path`?

**Comment by alexraju** [comment]
@jen20 Yes, I was able to ssh into the bastion or jump instance from my terminal using the key mentioned.
